src/renderer/src/app-state/command-palette/context.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useEffect, useState } from 'react';
+
+import { useKeyBindings } from '../../hooks';
+
+export type CommandPaletteContextType = {
+  isOpen: boolean;
+  openCommandPalette: () => void;
+  closeCommandPalette: () => void;
+};
+
+export const CommandPaletteContext = createContext<CommandPaletteContextType>({
+  isOpen: false,
+  openCommandPalette: () => {},
+  closeCommandPalette: () => {},
+});
+
+export const CommandPaletteStateProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
```

Comment:
Moved to a context so that the Electron menu item event can be handled in a central place in the app. Assuming that there is a single command palette open at a given time.